### PR TITLE
samples/cellular: update default libc

### PR DESCRIPTION
**Pull request by lemrey**
These samples needed to explicitly select `newlibc` when `minimal` libc was the default upstream.
Now that `picolibc` is the default, let samples uses the default C library where possible.

In some places I have removed float support in `printf` and `scanf` too, as it did not seem to be necessary.

### CODE REVIEW

**Comment by lemrey** [comment]
I kept out `cellular/download` and `cellular/https_client` since they are being moved, and `cellular/http_update` because it is being updated.

**Review by SeppoTakalo — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

File: samples/cellular/azure_fota/prj.conf [hunk 1]
```diff
@@ -67,8 +67,7 @@ CONFIG_NET_NATIVE=n
 CONFIG_NRF_MODEM_LIB=y
 
 # newlibc
```
Comment (tokangas) [inline samples/cellular/azure_fota/prj.conf:69]: This comment should be removed or updated.

**Review by tokangas — APPROVED** [review]

**Review by juhaylinen — APPROVED** [review]

File: samples/cellular/lwm2m_carrier/prj.conf [hunk 2]
```diff
@@ -7,7 +7,6 @@
 CONFIG_LWM2M_CARRIER=y
 
 # The library requires newlibc
```
[the remark this answers is not in the document]
Comment (lemrey) [reply samples/cellular/lwm2m_carrier/prj.conf]: thanks, updated

File: samples/cellular/azure_fota/prj.conf (hunk 1 above)
Comment (lemrey) [reply samples/cellular/azure_fota/prj.conf]: Thanks, I have updated that

**Review by rlubos — APPROVED** [review]

**Pull request merged**